pyproject.toml

```diff
@@ -24,27 +24,28 @@ license = "Apache License V 2.0"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = ">=3.8.0"
-pyspark = ">=3.3.0,<=3.4.0"
+python = ">=3.12.0"
+pyspark = ">=3.5.0,<=3.6.0"
 numpy = ">=1.16.4"
 s3fs = "2022.11.0"
 boto3 = "1.24.59"
 pandas = ">=1.4.2"
 great-expectations = "0.16.15"
-scikit-learn = "1.1.1"
+scikit-learn = "1.4.0"
 alphabet-detector = "0.0.7"
-pyarrow = "7.0.0"
-tidal-per-transformers = ">=0.0.8"
-torch = "1.12.1"
-mlflow = "2.1.1"
+pyarrow = "15.0.0"
+tidal-per-transformers = ">=0.1.0"
+torch = "2.2.1"
```

Comment:
@matt-taoma  this might cause some issues for saved models, let's be careful when we deploy this in downstream repos